### PR TITLE
Implement onSelectSeat callback in the js API.

### PR DESCRIPTION
**Pull request by zergov**
This PR implements the `onSelectSeat` callback function usable via the javascript API.
It lets the user define a function to be called when the user clicks on a seat in the floorplan.

Example:
```js
    const floorplan = new floorplanets.Floorplan({
      div: 'root',
      
      // seat parameter contains informations about the selected seat.
      onSelectSeat: (seat) => {
        console.log(seat);
      },

    });
```

The seat parameter object is defined [here](https://github.com/lanets/floorplanets/blob/delegate_onSelect_to_client/src/types.js#L21)

### CODE REVIEW

**Comment by zergov** [comment]
@aviau review pls <3

File: public/index.html [hunk 1]
```diff
@@ -43,6 +43,9 @@
      */
     const floorplan = new floorplanets.Floorplan({
       div: 'root',
+      onSeatSelected: (seat) => {
```
Comment (ppot) [inline public/index.html]: `onSelectedSeat` is a better naming

File: src/components/Floorplan.js [hunk 2]
```diff
@@ -21,16 +26,24 @@ export default class Floorplan extends React.Component {
...
       const seatdata = this.props.seats[id];
       const seat = new Seat(seatdata.x, seatdata.y, seatdata.label);
-      seats.push(seat);
+
+      seat.onSelect = () => this.handleSelectSeat(id);
     }
+  }
 
-    console.log('Floorplan initialized.');
+  handleSelectSeat(id: string) {
```
Comment (ppot) [inline src/components/Floorplan.js]: `handleSelectedSeat ` is a better naming

Comment (zergov) [inline src/components/Floorplan.js]: I don't agree on this one, 
`handleSelectedSeat` means it's already selected. The action of the selection should happen INSIDE this function and not before.

File: public/index.html (hunk 1 above)
Comment (zergov) [inline public/index.html]: You're right, the verb should come before.

File: src/components/Floorplan.js (hunk 2 above)
Comment (ppot) [inline src/components/Floorplan.js]: ya, just went with the previous for same naming ;) your right

File: __tests__/conversions.test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { toSeatData } from '../src/conversions';
+
+describe('toSeatData', () => {
```
Comment (Shuhala) [inline __tests__/conversions.test.js]: Just a general tip. Usually when unit testing, your descriptions should be more verbose.

In this case, describe could be something like "When using conversions.." (I didn't read your entire project so maybe there's a better description to write than this one). This way, your it functions should complete this sentence.

Your code will then look like this (below) and will be easier to read and understand, especially when unit testing more complex components.

describe("When doing something.."){
   it (' should do this ');
   it (' should do that ');
}

Output: 
When doing something..
  should do this..
  should do that..

Comment (zergov) [inline __tests__/conversions.test.js]: You're right @Shuhala :) Ill change it right away !

Comment (zergov) [inline __tests__/conversions.test.js]: @Shuhala Done 7d853370026ec50df29a4b3f0ac636ed160e1a14

**Pull request merged**